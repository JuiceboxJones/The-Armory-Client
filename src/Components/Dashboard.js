import React, { useState, useEffect, useContext } from 'react';
import TokenService from '../services/token-service';
import config from '../config';
import GamesContext from '../Contexts/gamesContext';

export default function Dashboard() {

  const games = useContext(GamesContext);

  function getGames() {
    return fetch(`${config.API_ENDPOINT}/games`, {
      headers: {
        authorization: `Bearer ${TokenService.getAuthToken()}`
      }
    }).then(res => (!res.ok ? TokenService.clearAuthToken() : res.json()));
  }

  useEffect(() => {
    getGames().then(data => {
      games.setGamesList(data);
    })
  },[]);

  const staticData = [
    {
      title: 'Overwatch',
      tags: ['FPS', 'Team-based', 'Hero-shooter'],
      AvailParties: 31,
      img:
        'https://upload.wikimedia.org/wikipedia/commons/1/1b/Square_200x200.png'
    },
    {
      title: 'FFXIV',
      tags: ['MMORPG'],
      AvailParties: 26,
      img:
        'https://upload.wikimedia.org/wikipedia/commons/1/1b/Square_200x200.png'
    },
    {
      title: 'League of Legends',
      tags: ['MOBA', 'Team-based', 'Strategy'],
      AvailParties: 78,
      img:
        'https://upload.wikimedia.org/wikipedia/commons/1/1b/Square_200x200.png'
    },
    {
      title: 'Apex Legends',
      tags: ['FPS', 'Battle Royale', 'Team-based', 'Hero-shooter'],
      AvailParties: 49,
      img:
        'https://upload.wikimedia.org/wikipedia/commons/1/1b/Square_200x200.png'
    },
    {
      title: 'CSGO',
      tags: ['FPS', 'Team-based'],
      AvailParties: 13,
      img:
        'https://upload.wikimedia.org/wikipedia/commons/1/1b/Square_200x200.png'
    },
    {
      title: 'DOTA 2',
      tags: ['MOBA', 'Strategy', 'PvP'],
      AvailParties: 55,
      img:
        'https://upload.wikimedia.org/wikipedia/commons/1/1b/Square_200x200.png'
    },
    {
      title: 'Fortnite',
      tags: ['Third-person shooter', 'Team-based', 'Battle Royale'],
      AvailParties: 3,
      img:
        'https://upload.wikimedia.org/wikipedia/commons/1/1b/Square_200x200.png'
    },
    {
      title: 'Rainbow Six Siege',
      tags: ['FPS', 'Team-based', 'Hero-shooter'],
      AvailParties: 67,
      img:
        'https://upload.wikimedia.org/wikipedia/commons/1/1b/Square_200x200.png'
    }
  ];

  const [filter, setFilter] = useState({ filter: 'all' });

  function handleData(staticData) {
    return staticData.map((data, index) => {
      return (
        <div key={index}>
          <img src={data.img} alt="Game Cover" />
          <h3>{data.title}</h3>
          {data.tags.map((tag, i) => {
            return <span key={i}>{tag} | </span>;
          })}
        </div>
      );
    });
  }

  return (
    <div className="dashboard-container">
      <ul>{handleData(staticData)}</ul>
      {/* <p>{filter}</p>
  <button onClick={e => setFilter({filter: 'all'})}>All Games</button>
  <button onClick={e => setFilter({filter: 'all'})}>All Games</button> */}
    </div>
  );
}
