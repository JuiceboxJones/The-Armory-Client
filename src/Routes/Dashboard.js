import React, { useState, useEffect, useContext } from "react";
import GamesContext from "../Contexts/gamesContext";
import Nav from "../Components/Nav";
import Game from "../Components/Game";
import GamesApiService from "../services/games-api-service";

export default function Dashboard(props) {
  const games = useContext(GamesContext);
  const staticData = games.gamesList;
  const [filter, setFilter] = useState("All");

  useEffect(() => {
    GamesApiService.getGames().then(data => {
      games.setGamesList(data);
    });
  }, []);

  const submitSearch = e => {
    e.preventDefault();
    const { search } = e.target;
    const params = { query: search.value };
    console.log("query: search.value =", params);
    GamesApiService.getGamesByTitle(params).then(data => {
      games.setGamesList(data);
    });
  };

  function displayGamesList(staticData) {
    console.log("staticData =", staticData);
    if (staticData.length === 0) {
      return (
        <span className="game-search-no-result">
          Game not found, please try again!
        </span>
      );
    } else {
      return staticData.map(props => {
        return <Game props={props} />;
      });
    }
  }

  // filter for tags
  function filterGames(staticData, keyword) {
    return staticData.map((props, index) => {
      return props.tags.includes(keyword) ? (
        <Game props={props} />
      ) : (
        <div key={index} />
      );
    });
  }

  return (
    <div className="dashboard-container">
      <Nav props={props} />
      <form className="game-search-form" onSubmit={submitSearch}>
        <input
          id="title-search-input"
          type="text"
          name="search"
          placeholder="Enter game title"
        />
        <button type="submit" className="games-search-button">
          Search
        </button>
      </form>
      {/* {displayGamesList(staticData)} */}
      <button className="filterGames" onClick={e => setFilter("All")}>
        All
      </button>
      <button className="filterGames" onClick={e => setFilter("Shooter")}>
        Shooter
      </button>
      <button className="filterGames" onClick={e => setFilter("MOBA")}>
        MOBA
      </button>
      <button className="filterGames" onClick={e => setFilter("MMORPG")}>
        MMORPG
      </button>
      <br/>
      <p>Filtering by: {filter}</p>
      <ul>
        {filter === "All"
          ? displayGamesList(staticData)
          : filterGames(staticData, filter)}
      </ul>
    </div>
  );
}
