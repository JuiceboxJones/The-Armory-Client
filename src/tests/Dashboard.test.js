import React from 'react';
import Dashboard from '../Routes/Dashboard';
import renderer from "react-test-renderer";
import { MemoryRouter } from 'react-router-dom';
import { shallow } from "enzyme";


describe('Dashboard component', () => {

  it('renders without crashing', () => {
    shallow(<Dashboard />)
  })

  it("renders the UI as expected", () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Dashboard />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
})