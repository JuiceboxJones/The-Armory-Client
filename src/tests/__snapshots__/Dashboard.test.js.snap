// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard component renders the UI as expected 1`] = `
<div
  className="dashboard-container"
>
  <div>
    <div>
      <img
        alt="avatar profile pic"
        src="basicAvatar.png"
      />
      <h2 />
      <button
        onClick={[Function]}
        type="button"
      >
        Edit Profile
      </button>
      <button
        onClick={[Function]}
      >
        Logout
      </button>
    </div>
  </div>
  <form
    className="game-search-form"
    onSubmit={[Function]}
  >
    <input
      id="title-search-input"
      name="search"
      placeholder="Enter game title"
      type="text"
    />
    <button
      className="games-search-button"
      type="submit"
    >
      Search
    </button>
  </form>
  <button
    className="filterGames"
    onClick={[Function]}
  >
    All
  </button>
  <button
    className="filterGames"
    onClick={[Function]}
  >
    Shooter
  </button>
  <button
    className="filterGames"
    onClick={[Function]}
  >
    MOBA
  </button>
  <button
    className="filterGames"
    onClick={[Function]}
  >
    MMORPG
  </button>
  <br />
  <p>
    Tag filter: 
    All
  </p>
  <ul>
    <span
      className="game-search-no-result"
    >
      Game not found, please try again!
    </span>
  </ul>
</div>
`;
